src/commands/generate/fromTemplate.ts

```diff
@@ -213,14 +227,33 @@ export default class Template extends Command {
     return { asyncapi, template, output };
   }
 
-  private parseFlags(disableHooks?: string[], params?: string[], mapBaseUrl?: string): ParsedFlags {
+  private parseFlags(disableHooks?: string[], params?: string[], mapBaseUrl?: string, registryUrl?: string, registryAuth?:string, registryToken?:string): ParsedFlags {
     return {
       params: this.paramParser(params),
       disableHooks: this.disableHooksParser(disableHooks),
       mapBaseUrlToFolder: this.mapBaseURLParser(mapBaseUrl),
+      registryURLValidation: this.registryURLParser(registryUrl),
+      registryAuthentication: this.registryValidation(registryUrl, registryAuth, registryToken)
+      
     } as ParsedFlags;
   }
 
+  // Parsing the url
+  private registryURLParser(input?:string) {
+    if (!input) { return; }
+    const isURL = /^https?:/;
+    if (!isURL.test(input.toLowerCase())) {
+      throw new Error('Invalid --registry-url flag. The param requires a valid http/https url.');
+    }
+  }
+  // No we need at least one parameter to be passed registryAuth and registryToken 
+
+  private registryValidation(registryUrl?:string, registryAuth?:string, registryToken?:string) {
+    if (registryUrl!=='https://registry.npmjs.org'&&registryAuth&&!registryToken) {
```

Comment:
What is it's a public registry other than npm?

Reply:
We can make a simple request to registry if it returns bad request, we can assume the registry is private